iac_validate/validator.py

```diff
@@ -92,9 +92,11 @@ def _validate_file(file_path: str) -> bool:
                     yamale.validate(self.schema, data, strict=True)
                 except YamaleError as e:
                     error = True
-                    msg = e.message.split("\n\t")[1]
-                    logger.error(msg)
-                    self.errors.append(msg)
+                    for result in e.results:
+                        print("Error validating data '%s' with '%s'\n\t" % (result.data, result.schema))
```

Comment:
Is there a specific reason you added the `print()`? Is the logging message not sufficient?

Reply:
Current implementation just prints syntax errors and then semantic errors without any separation. My intent was to add information to CLI output that "these are syntax errors" and "these are semantic errors", so the output will be more readable.
No specific reason for `print()`, It could be `logger.error("...")`.

Reply:
Current version:
```
iac-validate -s schema.yaml -r rules nxos_model_example_with_errors.yaml
ERROR - vlans.3.blah: Unexpected element
ERROR - Rule 103.001: BGP peer template is not configured under root.bgp.template_peers (['root.bgp.vrfs.0.neighbors.0 - SPINE-PEERS2', 'root.bgp.vrfs.0.neighbors.1 - SPINE-PEERS3'])
```

Intent:
```
Error validating data 'nxos_model_example_with_errors.yaml' with 'schema.yaml'

ERROR - vlans.3.blah: Unexpected element

Error in semantic validation:
ERROR - Rule 103.001: BGP peer template is not configured under root.bgp.template_peers (['root.bgp.vrfs.0.neighbors.0 - SPINE-PEERS2', 'root.bgp.vrfs.0.neighbors.1 - SPINE-PEERS3'])
```

Reply:
Understood. I would suggest to keep a single log message per error but expand the log message. Maybe something like this for syntax errors:
```
msg = "Syntax error '{}': {}".format(result.data, msg)
```
and this for semantic errors:
```
msg = "Semantic error, rule {}: {} ({})".format(id, self.rules[id].description, paths)
```

Reply:
Done.
Also fixed #2